### PR TITLE
[AArch64][PAC][MC][ELF] Support PAuth ABI compatibility tag

### PR DESCRIPTION
**Pull request by kovdan01**
Depends on #87545

Emit `GNU_PROPERTY_AARCH64_FEATURE_PAUTH` property in `.note.gnu.property` section depending on
`aarch64-elf-pauthabi-platform` and `aarch64-elf-pauthabi-version` llvm module flags.

### CODE REVIEW

**Comment by kovdan01** [comment]
Buildkite job expectedly ails since the PR depends on #85235 unmerged yet

File: llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp [hunk 1]
```diff
@@ -268,13 +268,24 @@ void AArch64AsmPrinter::emitStartOfAsmFile(Module &M) {
...
+  uint64_t PAuthABIPlatform = -1;
+  if (const auto *PAP = mdconst::extract_or_null<ConstantInt>(
+          M.getModuleFlag("aarch64-elf-pauthabi-platform")))
+    PAuthABIPlatform = PAP->getZExtValue();
+  uint64_t PAuthABIVersion = -1;
+  if (const auto *PAV = mdconst::extract_or_null<ConstantInt>(
+          M.getModuleFlag("aarch64-elf-pauthabi-version")))
+    PAuthABIVersion = PAV->getZExtValue();
+
+  if ((PAuthABIPlatform == uint64_t(-1)) != (PAuthABIVersion == uint64_t(-1)))
```
Comment (DanielKristofKiss) [inline llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp:280]: could this check be added to the verifier?

Comment (kovdan01) [reply llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp]: Thanks for suggestion, see ab89fdcb49f9ae6dd2d20cac81cca5204eda51a6

File: llvm/lib/IR/Verifier.cpp [hunk 2]
```diff
@@ -1734,8 +1734,29 @@ void Verifier::visitModuleFlags() {
...
+      } else if (FlagName->getString() == "aarch64-elf-pauthabi-version") {
+        if (const auto *PAV =
+                mdconst::dyn_extract_or_null<ConstantInt>(MDN->getOperand(2)))
+          PAuthABIVersion = PAV->getZExtValue();
+      }
+    }
+  }
+
+  if ((PAuthABIPlatform == uint64_t(-1)) != (PAuthABIVersion == uint64_t(-1)))
+    report_fatal_error(
```
Comment (DanielKristofKiss) [inline llvm/lib/IR/Verifier.cpp:1757]:
```suggestion
    CheckFailed(
```

**Review by DanielKristofKiss — APPROVED** [review]
Could you consider adding a test for the verifier?

**Review by MaskRay — APPROVED** [review]

File: llvm/lib/IR/Verifier.cpp (hunk 2 above)
Comment (kovdan01) [reply llvm/lib/IR/Verifier.cpp]: Fixed, thanks, see 2f6dc0adbee9e09176e86dd9960b8d353c31decf

**Comment by kovdan01** [comment]
> Could you consider adding a test for the verifier?

@DanielKristofKiss Thanks for suggestion, see llvm/test/Verifier/module-flags-note-gnu-property-elf-pauthabi.ll added in 2f6dc0adbee9e09176e86dd9960b8d353c31decf

**Pull request merged**